### DISCUSSION TITLE
RandCropByPosNegLabel: Random location

### DISCUSSION THREAD

**Discussion by lukasfolle** [Project-MONAI/MONAI#2115] [Ideas]
**Is your feature request related to a problem? Please describe.**
I noticed that `RandCropByPosNegLabel` always crops objects with the foreground pixels being in the center. 

**Describe the solution you'd like**
To get more variance into this transformation, I think it would be nice to have a flag indicating that the foreground voxel has not to be sampled in a centered way, but randomly in the patch. Especially for segmentation task with very few foreground voxels this would make sense IMO.

I am not entirely sure how that would affect model experiements but I would like to find out in case this gets implemented.
BTW: I absolutely love the functionality that this transformation already has!

**Comment by Nic-Ma** [comment]
Hi @lukasfolle ,

Thanks for raising this up.
Actually, we have a duplicated ticket: https://github.com/Project-MONAI/MONAI/issues/586
Could you please help submit a PR for this new transform if you are interested?

Thanks.

**Comment by wyli** [comment]
i think this could be achieved by providing a dilated version of the original foreground as the sampling candidate locations (label_key).  it’s slightly different from #586

**Comment by lukasfolle** [comment]
@wyli Could you elaborate on "dilated version of the original foreground as the sampling candidate locations"?

I had the following in mind:
1. Get foreground indices like before
2. Create a region around the foreground indices such that every valid crop yields a least one foreground index
3. Randomly sample the region

Please let me know what you think of this proposal.

**Comment by wyli** [comment]
yes, to achieve the same without modifying the transform:
1. dilate the foreground by k voxels and put it into the data dict with "label_dilated" key
2. invoke RandCropByPosNegLabel (label_key="label_dilated", spatial_size=2*k, ...)

**Comment by lukasfolle** [comment]
I agree on 1., but why should the spatial_size be changed in any way? If we choose k appropriately (depending on foregorund extent), shouldn´t the spatial_size be kept the same?

**Comment by wyli** [comment]
sure, dilation k and spatial size are free parameters here, I just wanted to set them in this particular relation so that the sampling roughly achieves "not to be sampled in a centered way, but randomly in the patch."

**Comment by lukasfolle** [comment]
@wyli Do you have some code working already?

**Comment by lukasfolle** [comment]
bump
